searchlib/src/vespa/searchlib/queryeval/same_element_blueprint.cpp

```diff
@@ -56,8 +56,15 @@ SameElementBlueprint::optimize_self()
 void
 SameElementBlueprint::fetchPostings(const ExecuteInfo &execInfo)
 {
-    for (size_t i = 0; i < _terms.size(); ++i) {
-        _terms[i]->fetchPostings(ExecuteInfo::create(execInfo.isStrict() && (i == 0), execInfo));
+    if (_terms.empty()) return;
```

Comment:
I do not think 2x code is worth the one call to hit_ratio

Reply:
I did it for readability. I find it better to handle the first outside when it is different than the next. Here it is strictness that differs.

Reply:
I see, but you could still get rid of the current/prev shuffle if you called hit_ratio one extra time...